exercises/practice/sum-of-multiples/.approaches/from-factors/content.md

```diff
@@ -0,0 +1,54 @@
+# Calculating sum from factors
+
+```rust
+pub fn sum_of_multiples_from_factors(limit: u32, factors: &[u32]) -> u32 {
+    let mut multiples: Vec<_> = factors
+        .iter()
+        .filter(|&&factor| factor != 0)
+        .flat_map(|&factor| (factor..limit).step_by(factor as usize))
+        .collect();
+    multiples.sort();
+    multiples.dedup();
+    multiples.iter().sum()
+}
+```
+
+This approach implements the exact steps outlined in the exercise description:
+
+1. For each non-zero factor, find all multiples of that factor that are less than the `limit`
+2. Collect all multiples in a [`Vec`][vec]
+3. Remove duplicate multiples
+3. Calculate the sum of all unique multiples
+
+In order to compute the list of multiples for a factor, we create a [`Range`][range] from the factor (inclusive) to the `limit` (exclusive), then use [`step_by`][iterator-step_by] with the same factor.
+
+To combine the multiples of all factors, we iterate the list of factors and use [`flat_map`][iterator-flat_map] on each factor's multiples.
+[`flat_map`][iterator-flat_map] is a combination of [`map`][iterator-map] and [`flatten`][iterator-flatten]; it maps each factor into its multiples, then flattens them all in a single sequence.
+
+Since we need to have unique multiples to compute the proper sum, we [`collect`][iterator-collect] the multiples into a [`Vec`][vec], which allows us to then [`sort`][slice-sort][^1] them and use [`dedup`][vec-dedup] to remove the duplicates.
+[`collect`][iterator-collect] is a powerful function that can collect the data in a sequence and store it in any kind of collection - however, because of this, the compiler is not able to infer the type of collection you want as the output.
+To solve this problem, we type the `multiples` variable explicitly.
+
+Finally, calculating the sum of the remaining unique multiples in the set is easy: we can simply call [`sum`][iterator-sum].
+
+[^1]: There is another method available to sort a slice: [`sort_unstable`][slice-sort_unstable]. Usually, using [`sort_unstable`][slice-sort_unstable] is recommended if we do not need to keep the ordering of duplicate elements (which is our case). However, [`sort`][slice-sort] has the advantage because of its implementation. From the documentation:
```

Comment:
Footnotes do not render properly: [this document](https://exercism.org/tracks/rust/exercises/sum-of-multiples/approaches/from-factors). Clicking the to/from the footnote leads to the top of the page.

Reply:
Good catch, thanks. I was wondering about this as well and saw that footnotes are in fact used [here](https://exercism.org/tracks/rust/concepts/structs). Forgot to check again after deployment.

What could be the problem? maybe the missing line `---`? I might ask on discord, someone's bound to know the answer.

Reply:
Actually, on the site I linked, the hyperlinks to/from footnotes don't work either. This might be a general problem of the website.

Reply:
posted a bug report on the forum https://forum.exercism.org/t/footnote-links-dont-work/9331